src/main/java/hudson/plugins/gradle/injection/GradleBuildScanInjection.java

```diff
@@ -86,59 +75,59 @@ private static String getInitScriptDirectory(EnvVars envGlobal, EnvVars envCompu
 
     private void inject(InjectionConfig config, Node node, String initScriptDirectory) {
         try {
-            EnvUtil.setEnvVar(node, JENKINSGRADLEPLUGIN_GRADLE_AUTO_INJECTION, "true");
-            injectInitScript(node.getChannel(), initScriptDirectory);
+            EnvUtil.setEnvVar(node, InitScriptVariables.DEVELOCITY_INJECTION_ENABLED, "true");
+            injectInitScript(node, initScriptDirectory);
             injectEnvironmentVariables(config, node);
         } catch (IOException | InterruptedException e) {
             throw new IllegalStateException(e);
         }
     }
 
-    private void injectInitScript(VirtualChannel channel, String initScriptDirectory) throws IOException, InterruptedException {
-        FilePath gradleInitScriptFile = getInitScriptFile(channel, initScriptDirectory);
+    private void injectInitScript(Node node, String initScriptDirectory) throws IOException, InterruptedException {
+        FilePath gradleInitScriptFile = getInitScriptFile(node.getChannel(), initScriptDirectory);
         if (initScriptChanged(gradleInitScriptFile)) {
             LOGGER.info("Injecting Gradle init script " + gradleInitScriptFile);
-
-            copyResourceToNode(gradleInitScriptFile, RESOURCE_INIT_SCRIPT_GRADLE);
+            writeToNode(gradleInitScriptFile, initScript.get());
```

Comment:
Because the script is templated and doesn't exist in the final form on disk or in a jar. Related to this comment https://github.com/jenkinsci/gradle-plugin/pull/409#discussion_r1528328438